src/value/owned.rs

```diff
@@ -357,110 +350,169 @@ impl Element for OwnedElement {
-        #[with(OwnedValue::Integer(AnyInt::I64(100)))] owned_element: OwnedElement,
-    ) -> IonResult<()> {
-        assert_eq!(&AnyInt::I64(100).as_i64(), &owned_element.as_i64());
-        assert_eq!(&AnyInt::I64(100).as_big_int(), &owned_element.as_big_int());
-        Ok(())
-    }
-
-    #[rstest]
-    fn test_as_float(
-        #[with(OwnedValue::Float(305e1))] owned_element: OwnedElement,
-    ) -> IonResult<()> {
-        assert_eq!(&Some(305e1), &owned_element.as_float());
-        Ok(())
-    }
-
-    #[rstest]
-    fn test_as_decimal(
-        #[with(OwnedValue::Decimal(Decimal::new(8, 3)))] owned_element: OwnedElement,
-    ) -> IonResult<()> {
-        let decimal_value2 = Decimal::new(80, 2);
-        let expected_elem = Some(&decimal_value2);
-
-        assert_eq!(&expected_elem, &owned_element.as_decimal());
-        Ok(())
+    /// Makes a timestamp from an RFC-3339 string and panics if it can't
+    fn mk_ts<T: AsRef<str>>(text: T) -> Timestamp {
+        DateTime::parse_from_rfc3339(text.as_ref()).unwrap().into()
     }
 
-    #[rstest]
-    fn test_as_timestamp(
-        #[with(OwnedValue::Timestamp(Timestamp::with_ymd_hms_millis(2021, 2, 5, 16, 43, 51, 192).clone().build_at_offset(5 * 60 * 60).unwrap().into()))]
-        owned_element: OwnedElement,
-    ) -> IonResult<()> {
-        let builder = Timestamp::with_ymd_hms_millis(2021, 2, 5, 16, 43, 51, 192);
-        let timestamp_value2 = builder.clone().build_at_offset(5 * 60 * 60)?;
-
-        assert_eq!(&Some(&timestamp_value2), &owned_element.as_timestamp());
-        Ok(())
-    }
+    // TODO consider refactoring in a common generic form for the `borrowed` module.
 
-    #[rstest]
-    fn test_as_str(
-        #[with(OwnedValue::String("hello".into()))] owned_element: OwnedElement,
-    ) -> IonResult<()> {
-        assert_eq!(&Some("hello"), &owned_element.as_str());
-        Ok(())
+    /// Models the operations on `Element` that we want to test.
+    #[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+    enum ElemOp {
+        IsNull,
+        AsBool,
+        AsAnyInt,
+        AsF64,
+        AsDecimal,
+        AsTimestamp,
+        AsStr,
+        AsSym,
+        AsBytes,
+        AsSequence,
+        AsStruct,
     }
 
-    #[rstest]
-    fn test_as_sym(
-        #[with(OwnedValue::Symbol("hello".into()))] owned_element: OwnedElement,
-    ) -> IonResult<()> {
-        let expected_elem: &OwnedSymbolToken = &("hello".into());
-
-        assert_eq!(
-            extract_text(expected_elem),
-            extract_text(owned_element.as_sym().unwrap().into())
-        );
-        Ok(())
-    }
-
-    #[rstest]
-    fn test_as_bool(
-        #[with(OwnedValue::Boolean(false))] owned_element: OwnedElement,
-    ) -> IonResult<()> {
-        assert_eq!(&Some(false), &owned_element.as_bool());
-        Ok(())
-    }
+    impl IntoIterator for ElemOp {
+        type Item = ElemOp;
+        type IntoIter = <Once<ElemOp> as IntoIterator>::IntoIter;
```

Comment:
The common case of using this is just replacing one operation--this allows us to use the single value as a singleton iterator in the fixture.